plugin-maven/src/main/java/com/diffplug/spotless/maven/ImpactedFilesTracker.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.maven;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Tracks the number of processed files, typically by a single Formatter for a whole repository
+ */
+public class ImpactedFilesTracker {
+	protected final AtomicInteger nbSkipped = new AtomicInteger();
+	protected final AtomicInteger nbChecked = new AtomicInteger();
+	protected final AtomicInteger nbCleaned = new AtomicInteger();
```

Comment:
A single `FormatterStep` does not support multithreading, so our `Formatter` doesn't either. I think regular integers with `++` is better to make it clear that the code is not actually threadsafe.